Mage/src/main/java/mage/abilities/effects/PayCostToAttackBlockEffectImpl.java

```diff
@@ -149,7 +146,9 @@ private boolean handleManaCosts(ManaCosts attackBlockManaTax, GameEvent event, A
             attackBlockManaTax.clearPaid();
             if (attackBlockManaTax.canPay(source, source.getSourceId(), player.getId(), game)
                     && player.chooseUse(Outcome.Neutral, chooseText, source, game)) {
-                handlePhyrexianManaCosts(manaCosts, player, source, game);
+                
+                handlePhyrexianManaCosts(getManaCostToPay(event, source, game), player, source, game);
```

Comment:
the change here - manaCosts was null at this point. but calling getManaCostToPay would retrieve the manaCosts calculated from the overridden getManaCostToPay methods in cards such as Sphere of Safety and Collective Restraint that were causing NREs